### ISSUE TITLE
@@-substitutions break validity of plugins' sources 

### ISSUE THREAD

**Issue by johnd0e** [IITC-CE/ingress-intel-total-conversion#50]
Plugin's sources now are not valid js code, because of build-time substitution blocks like `@@METAINFO@@`, `@@PLUGINSTART@@`, `@@PLUGINEND@@`.

It complicates applying such tools like `eslint`.

It would be great if we could rearrange the code to make the substitutions appear commented, e.g.
```
// @@METAINFO@@
// ...
// @@PLUGINSTART@@
// ...
// @@PLUGINEND@@
```

**Comment by modos189** [comment]
I think that if not now, then in the future we will still be forced to use nodejs in the project. And if so, then let's add it now. Can you connect nodejs, babel and eslint to our build system? You seem to have more experience with this than I do.

**Comment by johnd0e** [comment]
> And if so, then let's add it now.

Why not. But in fact it is beyond of scope of this issue: here we need very minor change to build script (I can make PR).

---
P.S
Yes, in the end it's about nodejs-dependent tools, but at first we need to be able to use them _locally_.
Later (but only after we have all the sources prepared) we also could include linting/prettying in our CI pipeline.

Currently we also need node.js+babel in our build process in order to resolve #47 keeping backward compatibility, but is is separate issue.

> You seem to have more experience with this than I do.

Actually I do not. But it seems not hard, so let's try)

**Comment by angel93ayora** [comment]
I've been building plugins with gulp and linting with eslint for a while, y simply inject code this way:
`/* inject: ./src/folder/file.js */`
or for html strings
```someHTMLElement.html(`/* inject ./src/html/file.html */`)```

i can send you the gulpfile and eslint files

**Comment by johnd0e** [comment]
> i can send you the gulpfile and eslint files

I'd like to see some samples, thanks in advance.

**Comment by angel93ayora** [comment]
i upload them here
[build-files.zip](https://github.com/IITC-CE/ingress-intel-total-conversion/files/2846945/build-files.zip)

take a look

**Comment by angel93ayora** [comment]
The gulpfile and eslint files would need to be changed to match folders, also gulpfile will need changes to build plugins separatedly and IITC.

**Comment by johnd0e** [comment]
@angel93ayora, thank you for samples.
I suppose we will transfer all our build system from python to node at some point.

As for this PR: I have new idea how we could completely get rid of all @@METAINFO/PLUGINSTART/PLUGINEND...
But it will be another PR.
